data_collection/gazette/spiders/sp_osasco.py

```diff
@@ -0,0 +1,57 @@
+import json
+import re
+from datetime import date
+
+import dateparser
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class SpOsascoSpider(BaseGazetteSpider):
+    TERRITORY_ID = "3534401"
+    name = "sp_osasco"
+    allowed_domains = ["www.osasco.sp.gov.br"]
+    start_urls = ["http://www.osasco.sp.gov.br/imprensa-oficial/"]
+    start_date = date(2002, 8, 2)
+
+    ONLY_JSON_REGEX = re.compile("(\\[.*}]);")
+    NUMBER_REGEX = re.compile("\\s([0-9]+)$")
+
+    def parse(self, response):
+        documents_json = response.xpath(
+            '//script[contains(text(), "DOCS")]'
+        ).extract_first()
+        match = self.ONLY_JSON_REGEX.search(documents_json)
+
+        if not match:
```

Comment:
**sugestão** Se não for possível extrair o JSON da página, não é necessário lançar essa exceção. Um erro estará nos logs e nenhhum item será retornado. No nosso monitoramente iremos perceber que algo está errado.

Reply:
Done